src/snowflake/cli/plugins/nativeapp/codegen/snowpark/models.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import List, Optional
+
+from pydantic import Field
+from snowflake.cli.api.project.schemas.snowpark.callable import _CallableBase
+from snowflake.cli.api.project.schemas.updatable_model import IdentifierField
+
+
+class ExtensionFunctionTypeEnum(str, Enum):
+    PROCEDURE = "PROCEDURE"
+    FUNCTION = "FUNCTION"
+    TABLE_FUNCTION = "TABLE_FUNCTION"
+    AGGREGATE_FUNCTION = "AGGREGATE_FUNCTION"
+
+
+class NativeAppExtensionFunction(_CallableBase):
+    function_type: ExtensionFunctionTypeEnum
+    lineno: Optional[int] = Field(
+        title="The line number of the extension function", default=None
+    )
+    name: Optional[str] = Field(title="The name of the external function", default=None)
+    packages: Optional[List[str]] = Field(
+        title="List of packages (with optional version constraints) to be loaded for the function",
```

Comment:
nit: should we specify these packages are already single quoted?

Reply:
I'm OK being a bit lenient here TBH